# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/windowwatcher/nsPIWindowWatcher.idl
# Commit: fde2d124a1b1
# Full Hash: fde2d124a1b15877570390c7c69f5fcf6770640d
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-08 03:36:50
# Regressor Bug: 1616353
# File Overlap Count: 1
# Description:
#   Bug 1616353 - Part 7.2: Create and use nsOpenWindowInfo types in nsWindowWatcher logic, r=kmag
#   
#   This patch builds on top of part 7.1 by creating this object within
#   nsWindowWatcher and ContentParent to carry the relevant information through
#   provider interfaces when creating new content windows. The nsOpenWindowInfo
# ==============================================================================

diff -r 7ccbbb127b7f -r fde2d124a1b1 toolkit/components/windowwatcher/nsPIWindowWatcher.idl
--- a/toolkit/components/windowwatcher/nsPIWindowWatcher.idl	Tue Apr 07 21:39:24 2020 +0000
+++ b/toolkit/components/windowwatcher/nsPIWindowWatcher.idl	Tue Apr 07 21:39:32 2020 +0000
@@ -21,6 +21,7 @@
 interface nsIDocShellTreeItem;
 interface nsIArray;
 interface nsIRemoteTab;
+interface nsIOpenWindowInfo;
 native nsDocShellLoadStatePtr(nsDocShellLoadState*);
 
 [uuid(d162f9c4-19d5-4723-931f-f1e51bfa9f68)]
@@ -106,12 +107,9 @@
    * @param aOpenerFullZoom
    *        The current zoom multiplier for the opener tab. This is then
    *        applied to the newly opened window.
-   * @param aNextRemoteTabId
-   *        The integer ID for the next remote tab actor.
-   *        0 means there is no next remote tab actor to use.
-   * @param aForceNoOpener
-   *        If true, then aOpeningTab will not be used to set the opener
-   *        for the newly created window.
+   * @param aOpenWindowInfo
+   *        Information used to create the initial content browser in the new
+   *        window.
    *
    * @return the nsIRemoteTab of the initial browser for the newly opened
    *         window.
@@ -120,7 +118,6 @@
                                        in ACString aFeatures,
                                        in boolean aCalledFromJS,
                                        in float aOpenerFullZoom,
-                                       in unsigned long long aNextRemoteTabId,
-                                       in boolean aForceNoOpener);
+                                       in nsIOpenWindowInfo aOpenWindowInfo);
 };
 